sdk/ai/azure-ai-inference/azure/ai/inference/models/_patch.py

```diff
@@ -0,0 +1,164 @@
+            return True
+        return self._deserialize_and_add_to_queue(element)
+
+    def __aiter__(self):
+        if not self._is_async_iterator:
+            raise ValueError("This method is only supported for async iterators")
+        return self
+
+    async def __anext__(self) -> _models.ChatCompletionsUpdate:
+        if not self._is_async_iterator:
+            raise ValueError("This method is only supported for async iterators")
+        while self._queue.empty() and not self._done:
+            self._done = await self._read_next_block_async()
+        if self._queue.empty():
+            raise StopAsyncIteration
+        return self._queue.get()
+
+    async def _read_next_block_async(self) -> bool:
+        try:
+            # Use 'cast' to make 'pyright' error go away
+            element = await cast(AsyncIterator[bytes], self._bytes_iterator).__anext__()
+        except StopAsyncIteration:
+            await self.aclose()
+            return True
+        return self._deserialize_and_add_to_queue(element)
+
+    def _deserialize_and_add_to_queue(self, element: bytes) -> bool:
+
+        # Clear the queue of ChatCompletionsUpdate before processing the next block
+        self._queue.queue.clear()
+
+        # Convert `bytes` to string and split the string by newline, while keeping the new line char.
+        # the last may be a partial "line" that does not contain a newline char at the end.
+        line_list: List[str] = re.split(r"(?<=\n)", element.decode("utf-8"))
+        for index, line in enumerate(line_list):
+
+            if self.ENABLE_CLASS_LOGS:
+                print(f"[original] {repr(line)}")
+
+            if index == 0:
+                line = self._incomplete_json + line
+                self._incomplete_json = ""
+
+            if index == len(line_list) - 1 and not line.endswith("\n"):
+                self._incomplete_json = line
+                return False
+
+            if self.ENABLE_CLASS_LOGS:
+                print(f"[modified] {repr(line)}")
+
+            if line == "\n":  # Empty line, indicating flush output to client
+                continue
+
+            if not line.startswith(self.SSE_DATA_EVENT_PREFIX):
+                raise ValueError(f"SSE event not supported (line `{line}`)")
+
+            if line.startswith(self.SSE_DATA_EVENT_DONE):
+                if self.ENABLE_CLASS_LOGS:
+                    print("done]")
+                return True
+
+            # If you reached here, the line should contain `data: {...}\n`
+            # where the curly braces contain a valid JSON object. Deserialize it into a ChatCompletionsUpdate object
+            # and add it to the queue.
+            self._queue.put(
+                # pylint: disable=W0212 # Access to a protected member _deserialize of a client class
+                _models.ChatCompletionsUpdate._deserialize(json.loads(line[len(self.SSE_DATA_EVENT_PREFIX) : -1]), [])
+            )
+
+            if self.ENABLE_CLASS_LOGS:
```

Comment:
No print statements please. If you want to log things (at a debug level), use [`logging`](https://docs.python.org/3/library/logging.html).

Reply:
@johanste I planned to remove these before PR complete, but instead I switched to using "logger" to log detailed SSE parsing. It's off by default since this is for development debugging. You need to update the code to enable these logs. Thanks!